contracts/protocol/libraries/logic/ValidationLogic.sol

```diff
@@ -252,6 +252,17 @@ library ValidationLogic {
             Errors.PRICE_ORACLE_SENTINEL_CHECK_FAILED
         );
 
+        if (params.releaseUnderlying) {
+            vars.availableLiquidity = IERC20(params.asset).balanceOf(
+                params.reserveCache.xTokenAddress
+            );
+
+            require(
+                vars.availableLiquidity >= params.amount,
+                Errors.NOT_ENOUGH_LIQUIDITY_IN_RESERVE
+            );
```

Comment:
if we already check collateral can always cover debt [here](https://github.com/para-space/paraspace-core/blob/da9d57291c01f33de106819347a624f4ae84b181/contracts/protocol/libraries/logic/ValidationLogic.sol#L327-L331) do we still need this?

Reply:
Yes, we need. what we are checking here is if cash is enough for borrowing, not LTV.

Reply:
yeah, basically we want to make sure that when the pool is releasing tokens to the users, that it actually has them.